executor/executor_test.go

```diff
@@ -8538,3 +8538,97 @@ func (s *testResourceTagSuite) TestResourceGroupTag(c *C) {
 		c.Assert(checkCnt > 0, IsTrue, commentf)
 	}
 }
+
+func (s *testStaleTxnSuite) TestStaleOrHistoryReadTemporaryTable(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	// For mocktikv, safe point is not initialized, we manually insert it for snapshot to use.
+	safePointName := "tikv_gc_safe_point"
+	safePointValue := "20160102-15:04:05 -0700"
+	safePointComment := "All versions after safe point can be accessed. (DO NOT EDIT)"
+	updateSafePoint := fmt.Sprintf(`INSERT INTO mysql.tidb VALUES ('%[1]s', '%[2]s', '%[3]s')
+	ON DUPLICATE KEY
+	UPDATE variable_value = '%[2]s', comment = '%[3]s'`, safePointName, safePointValue, safePointComment)
+	tk.MustExec(updateSafePoint)
+
+	tk.MustExec("set @@tidb_enable_global_temporary_table=1")
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists tmp1")
+	tk.MustExec("create global temporary table tmp1 " +
+		"(id int not null primary key, code int not null, value int default null, unique key code(code))" +
+		"on commit delete rows")
+	time.Sleep(time.Second)
```

Comment:
What's the purpose of this `Sleep()`? to make the CI slower?

Reply:
It was proposed for testing stale read with timestamp now() - interval 1 second, but found unnecessary at last. I'll remove it in next commit.

Reply:
If it runs to fast, timestamp NOW() will return a smaller ts than  ts which was used by create table because NOW() has a second precision. I'll remove time.Sleep and change NOW() to NOW(6)